lib/trino-orc/src/main/java/io/trino/orc/OrcReaderOptions.java

```diff
@@ -28,6 +28,7 @@
     private static final DataSize DEFAULT_MAX_BLOCK_SIZE = DataSize.of(16, MEGABYTE);
     private static final boolean DEFAULT_LAZY_READ_SMALL_RANGES = true;
     private static final boolean DEFAULT_NESTED_LAZY = true;
+    private static final int DEFAULT_EXPAND_DISCRETE_VALUES_LIMIT = 10;
```

Comment:
Why 10, actually? Document reasoning. Why not 100, 1000 or 10_000?

Reply:
I am actually not sure about the optimal value, so I took a conservative limit (even less that `Domain.DEFAULT_COMPACTION_THRESHOLD`).
We can set it to 0 - so by default this feature will be disabled. WDYT?

Reply:
Disabled by default is hard to justify here as it's unlikely that a user would think to enable this or figure out the right value by themselves.
Can we re-use the existing domain compaction threshold so that we expand a range only if the result stays under that threshold ?